src/main/java/edu/ucsb/courses/controllers/StatisticsController.java

```diff
@@ -228,4 +234,49 @@
         return ResponseEntity.ok().body(body);
     }
     
-}
+
+    private static String makeFormattedCourseName (
+        String department  ,
+        String courseNumber ,
+        String courseSuf    ) {
+
+        return
+              String.format( "%-8s", department                ) // 'CMPSC   '
+            + String.format( "%3s" , courseNumber               ) // '  8'
+            + String.format( "%-2s", courseSuf.toUpperCase()    ) // 'A '
+        ;
+    }
+
+    @GetMapping(value = "/singleCourseSearch", produces = "application/json")
+    public ResponseEntity<String> singleCourseSearch(@RequestParam(required = true) String startQuarter, @RequestParam(required = true) String endQuarter, 
+    @RequestParam(required = true) String department, @RequestParam(required = true) String courseNumber,@RequestParam(required = true) String courseSuf)
+            throws JsonProcessingException {
+
+        
+        List<Course> courseResults = courseRepository.findByQuarterIntervalAndCourseName (
+            startQuarter    ,
+            endQuarter      ,
+            makeFormattedCourseName(department, courseNumber, courseSuf)
+        );
+        
+        Map<String, String> professorInfo = new HashMap<>();
+
+        for(int i = 0; i < courseResults.size(); i++){
+            String instructor = courseResults.get(i).getClassSections().get(0).getInstructors().get(0).getInstructor();
+            if(professorInfo.get(instructor) != null){
+                int temp = Integer.parseInt(professorInfo.get(instructor));
+                professorInfo.put(instructor, Integer.toString(temp + 1));
+            
+
+            //New Professor
+            }else{
+                professorInfo.put(courseResults.get(i).getClassSections().get(0).getInstructors().get(0).getInstructor(), "1");
+            }   
+        }
+
+        String body = mapper.writeValueAsString(professorInfo);
+        
+        System.out.println(body);
```

Comment:
Generally, we want to reserve System.out.print that aid in debugging. Would we be able to remove this System.out.println?

Reply:
Issue fixed.